### DISCUSSION TITLE
Weird screen refreshing issue

### DISCUSSION THREAD

**Discussion by pauvilellam** [djphazer/O_C-Phazerville#106] [General]
Hey!

On my unit there is a strange screen issue. When turning knobs quickly or some applet is running which requires screen refreshing, small white dots and lines show up to the right of the screen. Please see videos attached. They can be seen to the right of the words "off" and "root" on one video and to the right of "G" and "D+1" on the other video. On one of them I am turning knobs which causes the white dots to show and on the other they show up on their own due to the screen refreshing. It happens across all applets. I have checked with the manufacturer an he has confirmed this happens across all the units so it must be software related. Would be awesome if this could be fixed in a future revision.
Thanks
Pau

Videos:
https://github.com/user-attachments/assets/02faf816-c9b3-4c9e-8f9a-1dd4a5a8a297
https://github.com/user-attachments/assets/5efc67d2-8b3c-462e-96ea-acae35fc6e92

**Comment by djphazer** [comment]
Indeed, I've noticed this on all Teensy 3.2 based units. It's relatively minor, doesn't seem to interfere with CV operation, so I've tried to ignore it ... and I think you're the first one to point it out. You can really see it in the debug menu (push left encoder on main menu).

I think it first appeared when I merged some code (from stock firmware v1.3.7-alpha) to use DMA for display updates. It's either a side effect of how fast the display is updated, or a sign of some small memory corruption on the edge of the framebuffer. It would certainly be nice to track it down!

**Reply by pauvilellam** [reply]
Yes, indeed. Would be nice to track it down if possible at some point. Anyways, thanks for comming back! (and for your hard work!)
Pau

**Comment by djphazer** [comment]
As a side effect of testing a larger 2.4" OLED screen, I now have a deeper understanding of how the display driver works. I think the glitches at the edge of the screen are a result of the SPI transfer being interrupted _sometimes_. I've merged in some code on a development branch that splits the display transfer up into smaller chunks, which helps prevent this. The tradeoff is slightly higher CPU usage (not a huge concern), and a slightly slower refresh rate (visibly noticeable, but minor).

Fresh builds for testing available here: https://github.com/djphazer/O_C-Phazerville/actions/runs/11681824832

**Reply by pauvilellam** [reply]
You are number 1. Will give it a try and come back. Thanks again! ;-)
